### PR TITLE
Secondary column support

### PR DESCRIPTION
**Pull request by mattgrill**
## Issue

https://github.com/jsdrupal/drupal-admin-ui/issues/242

## Screenshot / UI changes

create_recipe

_LOLOL_

## Testing instructions

- Are there two columns?

_Alright_, this will need some styling help. Additionally since the information about which fields are supposed to exist in the secondary column is not available via an API, the field names are hard coded.

### CODE REVIEW

File: src/components/05_pages/NodeForm/index.js [hunk 1]
```diff
@@ -145,66 +145,66 @@ class NodeForm extends React.Component {
     schema.properties.data.properties.attributes.properties[fieldName] ||
     schema.properties.data.properties.relationships.properties[fieldName];
 
+  createField = ({ fieldName, widget, inputProps }) => {
```
Comment (dawehner) [inline src/components/05_pages/NodeForm/index.js]: Super nice to extract that into its own helper function

File: src/components/05_pages/NodeForm/index.js [hunk 2]
```diff
@@ -145,66 +145,66 @@ class NodeForm extends React.Component {
...
+  };
+
   render() {
+    let result = null;
+    if (this.props.schema && this.props.uiSchema) {
+      const { right, left } = createUISchema(
+        this.props.uiSchema.fieldSchema,
+        this.props.uiSchema.formDisplaySchema,
+        this.props.widgets,
+        ['promote', 'status', 'sticky'],
```
Comment (dawehner) [inline src/components/05_pages/NodeForm/index.js]: I think hardcoding the second column is totally a pragmatic choice right now! 👍

File: src/utils/api/schema.js [hunk 3]
```diff
@@ -39,33 +44,42 @@ const createUISchema = (fieldSchema, formDisplaySchema, widgets) =>
...
+            ? fieldSchema[currentFieldName].attributes.settings
+            : {}),
+          ...(Object.prototype.hasOwnProperty.call(
+            formDisplaySchema,
+            currentFieldName,
+          )
+            ? formDisplaySchema[currentFieldName].settings
+            : {}),
+        };
+        acc[
```
Comment (dawehner) [inline src/utils/api/schema.js]: I'm wondering whether we should keep the existing createUiSchema function and instead have a second function which creates the left/right information out of an existing schema. ❓

File: src/components/02_atoms/FileUpload/FileUpload.js [hunk 4]
```diff
@@ -10,19 +10,13 @@ import api from './../../../utils/api/api';
 const Element = styled('div')`
   border: 2px dashed grey;
   border-radius: 3px;
-  padding: 80px;
-  width: 100%;
-  position: relative;
+  display: flex;
```
Comment (dawehner) [inline src/components/02_atoms/FileUpload/FileUpload.js]: Does someone know the state of flexbox IE 11 support?

File: src/components/05_pages/NodeForm/index.js [hunk 5]
```diff
@@ -145,78 +148,91 @@ class NodeForm extends React.Component {
...
+        ? this.onAttributeChange
+        : this.onRelationshipChange)(fieldName),
+      inputProps,
+    });
+  };
+
   render() {
+    let result = null;
+    if (this.props.schema && this.props.uiSchema) {
+      const { right, left } = sortUISchemaFields(
```
Comment (dawehner) [inline src/components/05_pages/NodeForm/index.js]: Thank you!

File: src/utils/api/schema.js [hunk 6]
```diff
@@ -68,4 +68,15 @@ const createUISchema = (fieldSchema, formDisplaySchema, widgets) =>
       return acc;
     }, []);
 
-export { createEntity, createUISchema };
+const sortUISchemaFields = (schema, secondaryColumnFields) =>
+  schema.reduce(
```
Comment (dawehner) [inline src/utils/api/schema.js]: Nice use of reduce!

File: src/components/02_atoms/FileUpload/FileUpload.js (hunk 4 above)
Comment (martinfrances107) [reply src/components/02_atoms/FileUpload/FileUpload.js]: https://caniuse.com/#feat=flexbox

Under the known issues tab - there is a good definition of "partial" support.

**Pull request merged**